src/login/LoginPage.jsx

```diff
@@ -70,13 +70,16 @@ class LoginPage extends React.Component {
   }
 
   shouldComponentUpdate(nextProps) {
-    if (this.props.loginFormData !== nextProps.loginFormData) {
-      if (nextProps.loginFormData) {
-        this.setState({
-          ...nextProps.loginFormData,
-        });
-        return false;
-      }
+    if (this.props.loginFormData !== nextProps.loginFormData && nextProps.loginFormData) {
```

Comment:
@syedsajjadkazmii Why we are adding same condition two times ?
`nextProps.loginFormData && nextProps.loginFormData)`

Reply:
@attiyaIshaque actually `this.props.loginFormData !== nextProps.loginFormData` will be evaluated first and then the result will be evaluated with ` && nextProps.loginFormData`. This is because of the operator precedence, Relational operators are evaluated before logical operators, as `!==` is the relational operator and `&&` is logical operator. 
Moreover, first condition is checking for the state updates and the second condition is just validating if `nextProps.loginFormData` is not null.

Reply:
Plus, I felt like there is no need for a second condition, so I removed that